apps/judicial-system/backend/src/app/modules/case/filters/case.filter.ts

```diff
@@ -2,228 +2,207 @@ import {
   CaseDecision,
   CaseState,
   CaseType,
-  hasCaseBeenAppealed,
   indictmentCases,
   InstitutionType,
-  investigationCases,
   isIndictmentCase,
-  restrictionCases,
   UserRole,
-  isExtendedCourtRole,
   isProsecutionUser,
+  isDistrictCourtUser,
+  isAppealsCourtUser,
+  isPrisonSystemUser,
+  isRestrictionCase,
+  isInvestigationCase,
+  CaseAppealState,
 } from '@island.is/judicial-system/types'
-import type { User, Case as TCase } from '@island.is/judicial-system/types'
+import type { User } from '@island.is/judicial-system/types'
 
 import { Case } from '../models/case.model'
 
-function getAllowedStates(
+function canProsecutionUserAccessCase(
+  theCase: Case,
   user: User,
-  institutionType?: InstitutionType,
-  caseType?: CaseType,
-): CaseState[] {
-  if (isProsecutionUser(user)) {
-    return [
+  forUpdate = true,
+): boolean {
+  // Check case type access
+  if (user.role === UserRole.PROSECUTOR) {
+    if (
+      !isRestrictionCase(theCase.type) &&
+      !isInvestigationCase(theCase.type) &&
+      !isIndictmentCase(theCase.type)
+    ) {
+      return false
+    }
+  } else if (!isIndictmentCase(theCase.type)) {
```

Comment:
Why are we checking !isIndictmentCase(theCase.type) twice?

Reply:
We are not really checking it twice, but rather doing different checks base on user role.

So basically:
- If you are a prosecutor and the case type is not restriction, investigation or indictment, then you do not have access. Currently, this means prosecutors have access to all cases, but if we add not types of cases, prosecutors will not automatically get access.
- If you are some other role (currently only representative) and the case type is not indictment, then you do not get access.